addons/mail/static/tests/systray/systray_messaging_menu_tests.js

```diff
@@ -839,5 +839,45 @@ QUnit.test('messaging menu widget: do not open chat window twice on preview clic
     messagingMenu.destroy();
 });
 
+QUnit.test('messaging menu widget: expand on thread preview', function (assert) {
+    assert.expect(4);
+
+    var taskMessage = [{
+        author_id: [1, "Demo"],
+        body: "<p>Office Design Project</p>",
+        res_id: 123,
+        model: 'some.res.model',
+        needaction: true,
+        needaction_partner_ids: [44],
+    }];
+    this.data['mail.message'].records =
+        this.data['mail.message'].records.concat(taskMessage);
+
+    var messagingMenu = new MessagingMenu();
+    testUtils.mock.addMockEnvironment(messagingMenu, {
+        services: this.services,
+        data: this.data,
+        session: {
+            partner_id: 44,
+        },
+        intercepts: {
+            do_action: function (ev) {
+                assert.step('do_action');
```

Comment:
I would add more assertion about this `do_action`, to ensure it opens the related document.

Reply:
changes done